src/Text/Trifecta/Rendering.hs

```diff
@@ -350,29 +395,38 @@ instance Reducer (Careted a) Rendering where
 
 instance Hashable a => Hashable (Careted a)
 
+-- | ANSI terminal style to render spans with.
 spanEffects :: [SGR]
 spanEffects  = [SetColor Foreground Dull Green]
 
-drawSpan :: Delta -> Delta -> Delta -> Lines -> Lines
-drawSpan s e d a
-  | nl && nh  = go (column l) (rep (max (column h - column l) 0) '~') a
-  | nl        = go (column l) (rep (max (snd (snd (bounds a)) - column l + 1) 0) '~') a
-  |       nh  = go (-1)       (rep (max (column h + 1) 0) '~') a
-  | otherwise = a
+drawSpan
+    :: Delta -- ^ Start of the region of interest
+    -> Delta -- ^ End of the region of interest
+    -> Delta -- ^ Currrent location
+    -> Lines -- ^ 'Lines' to add the rendering to
+    -> Lines
+drawSpan start end d a
+  | nearLo && nearHi = go (column lo) (rep (max (column hi - column lo) 0) '~') a
+  | nearLo           = go (column lo) (rep (max (snd (snd (bounds a)) - column lo + 1) 0) '~') a
+  |           nearHi = go (-1)        (rep (max (column hi + 1) 0) '~') a
+  | otherwise        = a
   where
     go = draw spanEffects 1 . fromIntegral
-    l = argmin bytes s e
-    h = argmax bytes s e
-    nl = near l d
-    nh = near h d
+    lo = argmin bytes start end
+    hi = argmax bytes start end
+    nearLo = near lo d
+    nearHi = near hi d
     rep = P.replicate . fromIntegral
 
--- |
--- > int main(int argc, char ** argv) { int; }
--- >                                    ^~~
```

Comment:
I'm not sure if outright removing these Haddocks outright is a good thing. At the very least, they do provide a (somewhat terse) description of what `addSpan` accomplishes.